code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -471,3 +471,84 @@
 		M.emote(pick("twitch","laugh","frown"))
 	..()
 	. = 1
+
+//I had to do too much research on this to make this a thing. Hopefully the FBI won't kick my door down.
+/datum/reagent/drug/ketamine
+	name = "Ketamine"
+	description = "A heavy duty tranquilizer found to also invoke feelings of euphoria, and assist with pain. Popular at parties and amongst small frogmen who drive Honda Civics."
+	reagent_state = LIQUID
+	color = "#c9c9c9"
+	metabolization_rate = 0.5 * REAGENTS_METABOLISM
+	addiction_threshold = 8
+	overdose_threshold = 16
+
+/datum/reagent/drug/ketamine/on_mob_metabolize(mob/living/L)
+	ADD_TRAIT(L, TRAIT_IGNOREDAMAGESLOWDOWN, type)
+	. = ..()
+
+/datum/reagent/drug/ketamine/on_mob_delete(mob/living/L)
+	REMOVE_TRAIT(L, TRAIT_IGNOREDAMAGESLOWDOWN, type)
+	. = ..()
+
+/datum/reagent/drug/ketamine/on_mob_life(mob/living/carbon/M)
+	//Friendly Reminder: Ketamine is a tranquilizer and will sleep you.
+	switch(current_cycle)
+		if(10)
+			to_chat(M, "<span class='warning'>You start to feel tired...</span>" )
+		if(11 to 25)
+			M.drowsyness += 1
+		if(26 to INFINITY)
+			M.Sleeping(60, 0)
+			. = 1
+	//Providing a Mood Boost
+	M.confused -= 3
+	M.jitteriness -= 5
+	M.disgust -= 3
+	//Ketamine is also a dissociative anasthetic which means Hallucinations!
+	M.hallucination = 5
+	. = ..()
+
+/datum/reagent/drug/ketamine/overdose_process(mob/living/M)
+	. = ..()
+	//Dissociative Anaesthetics? Overdosing? Time to dissociate hard.
+	var/obj/item/organ/brain/B = M.getorgan(/obj/item/organ/brain)
+	if(B.can_gain_trauma(/datum/brain_trauma/severe/split_personality, 5))
+		B.brain_gain_trauma(/datum/brain_trauma/severe/split_personality, 5)
+	M.hallucination += 10
+	//Uh Oh Someone is tired
+	if(prob(40))
+		if(!(HAS_TRAIT(M, TRAIT_IGNOREDAMAGESLOWDOWN)))
+			REMOVE_TRAIT(M, TRAIT_IGNOREDAMAGESLOWDOWN, type)
```

Comment:
Is he doesn't have that trait you are removing it?

Reply:
Because I am so galaxy brained I fucked up the logic and lost the ability to spell and grammar.